src/test/java/com/uid2/shared/auth/OperatorKeyTest.java

```diff
@@ -71,4 +71,34 @@ public void verifySiteIdPropIsOptionalForBackwardsCompatibility() {
         OperatorKey c = OperatorKey.valueOf(jo);
         Assert.assertNull(c.getSiteId());
     }
+
+    @Test
+    public void verifyRolesPropIsOptionalForBackwardsCompatibility() {
+        final String testJson = "    {\n" +
+                "        \"key\": \"test-admin-key\",\n" +
+                "        \"name\": \"admin@uid2.com\",\n" +
+                "        \"contact\": \"admin@uid2.com\",\n" +
+                "        \"created\": 1617149276,\n" +
+                "        \"disabled\": false\n" +
+                "    }";
+
+        JsonObject jo = (JsonObject) Json.decodeValue(testJson);
+        OperatorKey c = OperatorKey.valueOf(jo);
+        Assert.assertTrue(c.getRoles().isEmpty());
+    }
+    @Test
+    public void verifyRolesPropSetOptoutRole() {
+        final String testJson = "    {\n" +
+                "        \"key\": \"test-admin-key\",\n" +
+                "        \"name\": \"admin@uid2.com\",\n" +
+                "        \"contact\": \"admin@uid2.com\",\n" +
+                "        \"created\": 1617149276,\n" +
+                "        \"disabled\": false,\n" +
+                "        \"roles\": [ \"optout\" ]\n" +
+                "    }";
+
+        JsonObject jo = (JsonObject) Json.decodeValue(testJson);
+        OperatorKey c = OperatorKey.valueOf(jo);
+        Assert.assertTrue(c.getRoles().contains(Role.OPTOUT));
+    }
```

Comment:
maybe add another couple of simple tests that defines an operator with just OPERATOR role, and another with both OPTOUT and OPERATOR roles as they are the most likely combos in prod ?

Reply:
Tests added

Reply:
might wanna test the `public boolean hasRole(Role role)` calls in these 2 tests as well instead of just testing `getRoles`